tests/core/pyspec/eth2spec/test/merge/block_processing/test_process_execution_payload.py

```diff
@@ -0,0 +1,40 @@
+from eth2spec.test.helpers.execution_payload import build_empty_execution_payload
+from eth2spec.test.context import spec_state_test, expect_assertion_error, with_merge_and_later
+from eth2spec.test.helpers.state import next_slot
+
+
+def run_execution_payload_processing(spec, state, execution_payload, valid=True, execution_valid=True):
+    """
+    Run ``process_execution_payload``, yielding:
+      - pre-state ('pre')
+      - execution payload ('execution_payload')
+      - execution details, to mock EVM execution ('execution.yml', a dict with 'execution_valid' key and boolean value)
+      - post-state ('post').
+    If ``valid == False``, run expecting ``AssertionError``
+    """
+
+    yield 'pre', state
+    yield 'execution', {'execution_valid': execution_valid}
+    yield 'execution_payload', execution_payload
+
+    if not valid:
+        expect_assertion_error(lambda: spec.process_execution_payload(state, execution_payload))
+        yield 'post', None
+        return
+
+    spec.process_execution_payload(state, execution_payload)
+
+    yield 'post', state
+
+    # TODO: any assertions to make?
+
+
+@with_merge_and_later
+@spec_state_test
+def test_success_first_payload(spec, state):
+    next_slot(spec, state)
+    assert not spec.is_transition_completed(state)
```

Comment:
Maybe assert that it is a transition block too

Reply:
There is no block processing here, just the execution payload, so as-is it cannot be asserted.

Maybe we should start using the new `run_block_processing_to` helper to simulate the block behavior up to the execution-payload processing, I can discuss this with Mikhail for the next PR, but I prefer to keep this PR simple.

Reply:
Is this test case supposed to test the transition?

Reply:
`process_execution_payload` has been refactored to not care about the transition logic. So yes, it tests the empty header to a real header but it doesn't  test that the transition go/no-go logic works

Reply:
A nit: then we likely should not use transition spec helpers to check the pre-state.

Reply:
ah, but we want to test a range of tests/pre-states/inputs directly on this function. One of which when `state` has an empty header which maps to `not spec.is_transition_completed`. Even though `process_execution_payload` does not check this, we do want to ensure we have a test cast that is in this pre-condition.

If we don't use that helper, we'd want to check `state.latest_execution_header == ExecutionHeader()`. Or directly just set it.